### PR TITLE
JS-2757a: Add Pagination and Drop Down per_page toggle

### PR DESCRIPTION
**Pull request by theyoyojo**
Also change minimum username length from 4 to 1 at the point of registration

### CODE REVIEW

File: app/assets/javascripts/projects/show.js.coffee [hunk 1]
```diff
@@ -335,6 +328,11 @@ IS.onReady "projects/show", ->
     e.preventDefault()
     window.print()
 
+  $("select#display_n").change ->
+    console.log $("#display_n option:selected").val()
```
Comment (doug-salvati) [inline app/assets/javascripts/projects/show.js.coffee]: Remove this line

File: app/assets/javascripts/projects/show.js.coffee [hunk 2]
```diff
@@ -387,3 +385,5 @@ IS.onReady "projects/show", ->
     $('#tag-badge-textfield').val('')
     $('#tag-badge-form').hide()
     $('#tag-badge-add').show()
+
+
```
Comment (doug-salvati) [inline app/assets/javascripts/projects/show.js.coffee]: Remove extra space

File: app/assets/javascripts/projects/show.js.coffee (hunk 1 above)
Comment (doug-salvati) [inline app/assets/javascripts/projects/show.js.coffee]: Remove console.log

File: app/assets/javascripts/projects/show.js.coffee (hunk 2 above)
Comment (doug-salvati) [inline app/assets/javascripts/projects/show.js.coffee]: Remove extra space

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -453,6 +459,7 @@ def finishTemplateUpload
         redirect_to project_path(@project) and return
       end
       # Don't save fields until we know they are all valid
+
```
Comment (doug-salvati) [inline app/controllers/projects_controller.rb]: Remove newline

File: app/models/user.rb [hunk 4]
```diff
@@ -9,7 +9,7 @@ class User < ActiveRecord::Base
 
   include ActionView::Helpers::SanitizeHelper
 
-  validates :name, length: { minimum: 4, maximum: 70 }, format: {
+  validates :name, length: { minimum: 1, maximum: 70 }, format: {
```
Comment (doug-salvati) [inline app/models/user.rb]: Should have done this in another PR, but OK since it's so simple. Please mention both issues in your PR description.

File: app/controllers/projects_controller.rb [hunk 5]
```diff
@@ -62,7 +62,14 @@ def show
 
     @cloned_project = Project.select(:id, :user_id, :title).where(id: @project.cloned_from).first
     @liked_by_cur_user = Like.find_by_user_id_and_project_id(current_user, @project.id)
-    @data_sets = @project.data_sets.includes(:user).select('id', 'title', 'user_id', 'key', 'created_at', 'contributor_name').search(params[:search])
+    @all_data_sets = @project.data_sets.includes(:user).select('id', 'title', 'user_id', 'key', 'created_at', 'contributor_name').search(params[:search])
+
+    # The comparission validates the per_page display. The minimum and maximum are arbitrarily chosen but reasonable
```
Comment (doug-salvati) [inline app/controllers/projects_controller.rb]: While you're at it, might as well fix spelling

File: app/models/user.rb (hunk 4 above)
Comment (theyoyojo) [reply app/models/user.rb]: Yeah, that was a mistake with git branching but for that same reason of simplicity and the fact that we all spoke about it together, I didn't fix it

Comment (theyoyojo) [reply app/models/user.rb]: Will change description

File: app/controllers/projects_controller.rb (hunk 5 above)
Comment (theyoyojo) [reply app/controllers/projects_controller.rb]: Good idea

File: app/models/user.rb (hunk 4 above)
Comment (theyoyojo) [reply app/models/user.rb]: Description updated 4 hours ago
